pkg/component/watch.go

```diff
@@ -50,26 +50,31 @@ func addRecursiveWatch(watcher *fsnotify.Watcher, path string, ignores []string)
 		}
 		return nil
 	})
+	fmt.Printf("\n\nIgnores is %s\n\n", ignores)
 	for _, v := range folders {
 		ignore := false
 		for _, pattern := range ignores {
-			if matched, _ := regexp.MatchString(pattern, v); matched {
+			// Don't watch:
+			// 	1. The main src directory as its files and sub-directories will already be watched
+			//	2. Anything in ignores
+			if matched, _ := regexp.MatchString(pattern, v); matched || v == path {
 				ignore = true
 				break
 			}
 		}
 		if ignore {
 			glog.V(4).Infof("ignoring watch for %s", v)
 			continue
-		}
-		glog.V(4).Infof("adding watch on path %s", v)
-		err = watcher.Add(v)
-		if err != nil {
-			// Linux "no space left on device" issues are usually resolved via
-			// $ sudo sysctl fs.inotify.max_user_watches=65536
-			// BSD / OSX: "too many open files" issues are ussualy resolved via
-			// $ sysctl variables "kern.maxfiles" and "kern.maxfilesperproc",
-			return fmt.Errorf("error adding watcher for path %s: %v", v, err)
+		} else {
+			glog.V(4).Infof("adding watch on path %s", v)
+			err = watcher.Add(v)
+			if err != nil {
+				// Linux "no space left on device" issues are usually resolved via
+				// $ sudo sysctl fs.inotify.max_user_watches=65536
+				// BSD / OSX: "too many open files" issues are ussualy resolved via
+				// $ sysctl variables "kern.maxfiles" and "kern.maxfilesperproc",
+				return fmt.Errorf("error adding watcher for path %s: %v", v, err)
```

Comment:
May be odd to add this but I want your opinion, if this error occurs, should we link it to a GitHub doc that has the instructions regarding `sudo sysctl fs.inotify.max_user_watches=65536` as well as the Mac OS instructions?

Reply:
or may be even a project wiki page(I tend to prefer)

Reply:
I like the idea of putting troubleshooting doc in wiki